README.md
# Chevre Domain Library for Node.js

[![npm (scoped)](https://img.shields.io/npm/v/@chevre/domain.svg)](https://www.npmjs.com/package/@chevre/domain)
[![CircleCI](https://circleci.com/gh/chevre-jp/domain.svg?style=svg)](https://circleci.com/gh/chevre-jp/domain)
[![Coverage Status](https://coveralls.io/repos/github/chevre-jp/domain/badge.svg?branch=master)](https://coveralls.io/github/chevre-jp/domain?branch=master)
[![Dependency Status](https://img.shields.io/david/chevre-jp/domain.svg)](https://david-dm.org/chevre-jp/domain)
[![Known Vulnerabilities](https://snyk.io/test/github/chevre-jp/domain/badge.svg?targetFile=package.json)](https://snyk.io/test/github/chevre-jp/domain?targetFile=package.json)
[![npm](https://img.shields.io/npm/dm/@chevre/domain.svg)](https://nodei.co/npm/@chevre/domain/)

在庫管理サービスドメイン

## Table of contents

* [Usage](#usage)
* [Code Samples](#code-samples)
* [License](#license)

## Usage

```shell
npm install --save @chevre/domain
```

```Node.js
const chevre = require("@chevre/domain");
```

### Environment variables

| Name                           | Required | Value           | Purpose                           |
| ------------------------------ | -------- | --------------- | --------------------------------- |
| `ABORTED_TASKS_WITHOUT_REPORT` | false    |                 | Aborted task names without report |
| `DEBUG`                        | false    | chevre-domain:* | Debug                             |
| `COA_ENDPOINT`                 | false    |                 | COA API Settings                  |
| `COA_REFRESH_TOKEN`            | false    |                 | COA API Settings                  |
| `INFORM_PAYMENT_URL`           | false    |                 | Inform payment URL                |
| `INFORM_RESERVATION_URL`       | false    |                 | Inform reservation URL            |
| `INFORM_USE_RESERVATION_URL`   | false    |                 | Inform use reservation action URL |
| `LINE_NOTIFY_URL`              | false    |                 | LINE Notify Settings              |
| `LINE_NOTIFY_ACCESS_TOKEN`     | false    |                 | LINE Notify Settings              |

## Code Samples

Code sample are [here](https://github.com/chevre-jp/domain/tree/master/example).

## License

ISC


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/).

## Unreleased

### Added

### Changed

### Deprecated

### Removed

### Fixed

### Security

## v15.58.3 - 2021-07-20

### Changed

- 注文カスタマーインターフェースを顧客組織にも拡張

## v15.58.2 - 2021-07-19

### Changed

- 取引のIAgentをウェブアプリケーションにも拡張

## v15.58.1 - 2021-07-19

### Changed

- 所有権の所有者インターフェースをウェブアプリケーションにも拡張

## v15.58.0 - 2021-07-18

### Added

- 決済URL発行ファンクションを追加

### Changed

- IAMロール検索条件拡張
- 予約にissuedThroughを追加
- 予約のreservationForを最適化
- プロダクトのserviceTypeをサービスアウトプットに連携
- 予約のprogramMembershipUsed型をPermitに変更
- プロダクト検索条件拡張
- サービスアウトプット検索条件拡張
- 決済サービスのserviceOutputへの依存をserviceTypeへの依存に変更
- ペイメントカード登録時に作成される口座のtypeOfをAccountに固定
- ペイメントカード決済承認前に決済方法検証処理を追加
- 予約開始時にissuedByの指定があれば適用するように調整
- 注文カスタマーインターフェースをウェブアプリケーションにも拡張

## v15.57.0 - 2021-07-04

### Changed

- プロジェクトメンバー検索条件拡張

## v15.56.0 - 2021-07-03

### Changed

- 各リソースのプロジェクトID検索条件指定を最適化($exists: trueを削除)
- 予約の単価オファーを最適化
- オファーカタログ検索条件拡張

## v15.55.0 - 2021-06-25

### Changed

- プロダクト検索条件拡張

## v15.54.0 - 2021-06-24

### Changed

- GMOリクエストにタイムアウトを設定
- 全リソースのプロジェクトID検索条件を統一
- 予約取消のttts互換性維持対応を削除

## v15.53.0 - 2021-06-23

### Changed

- 顧客コレクションインデックス調整

## v15.52.1 - 2021-06-23

### Changed

- 顧客コードをidentifier->branchCodeと変更

## v15.52.0 - 2021-06-21

### Changed

- プロダクト検索条件拡張
- オファー検索条件拡張

## v15.51.0 - 2021-06-20

### Changed

- 資産取引番号からプロジェクトプレフィックスを除外

## v15.50.0 - 2021-06-15

### Changed

- 単価オファーの提供アイテムを最適化
- サービスアウトプット検索にprojectionを指定できるように調整

## v15.49.0 - 2021-06-13

### Changed

- サービスアウトプット検索条件拡張
- 予約取引開始時の適用メンバーシップ検証にアクセスコードと有効期間確認処理を追加

## v15.48.0 - 2021-06-12

### Changed

- 予約取引に使用メンバーシップを指定できるように調整
- 予約取引開始時に適用メンバーシップ区分検証処理を追加
- 予約取引開始パラメータを最適化
- オファー検索条件拡張

## v15.47.0 - 2021-06-04

### Added

- インボイスリポジトリを追加
- 決済方法リポジトリを追加
- 注文リポジトリを追加
- 経理レポートリポジトリを追加
- 旧売上レポートリポジトリを追加
- ウェブフックサービスを追加
- コードリポジトリを追加
- 所有権リポジトリを追加
- プロジェクトメンバーリポジトリを追加
- IAMロールリポジトリを追加
- 顧客リポジトリを追加
- IAMサービスを追加
- telemetryリポジトリを追加
- 取引リポジトリを追加
- cinerinoからリポジトリを移行
- credentialsをエクスポート
- utilサービスを追加
- errorHandlerをエクスポート
- 口座リポジトリを追加
- 口座アクションリポジトリを追加
- 口座取引リポジトリを追加
- Pecorino SDKをエクスポート
- 口座サービスを追加(@pecorino/domainから移行)
- 口座取引サービスを追加(@pecorino/domainから移行)

### Changed

- 予約ステータス変更の冪等性担保
- 予約取消取引開始パラメータからtransactionを削除
- イベントの座席検索に$projection指定を追加
- 予約取消取引開始前に予約ステータスが確定済かどうか確認するように調整
- セクションに座席が未登録の場合のイベント予約集計を調整
- 返金アクションにtotalPaymentDueをセット
- プロジェクトのイベント変更時設定を廃止
- cinerinoのMongoDBインデックスを移行
- 資産取引コレクションをrename
- cinerinoのアクション検索条件を継承
- 資産取引のネームスペースを変更
- 口座リポジトリ参照のPecorino依存を排除
- 口座アクションリポジトリ参照のPecorino依存を排除
- 通貨転送取引中止時に取引が存在しない場合に対応
- サービス登録開始前に特典付与先口座の存在を確認するように調整

### Fixed

- イベントの座席オファー検索におけるコーディングバグに対応

## v15.46.1 - 2021-03-21

### Changed

- USE_CHECK_PAY_ACTION_BEFORE_REFUND設定を削除

## v15.46.0 - 2021-03-18

### Added

- 決済サービスタイプにFaceToFaceを追加
- ペイメントサービスのプロバイダー認証情報にトークン認証コードを追加
- 決済後処理を追加
- 返金後処理を追加
- 予約にbrokerを追加
- グローバル設定に決済ステータス変更時設定を追加
- グローバル設定に予約使用時設定を追加
- 決済と返金後の通知アクションを追加

### Changed

- 返品手数料決済処理をGMO取引の取消→即時売上で実装するように調整
- 返金取引開始前に決済アクションの存在確認を追加
- 決済取引開始時に販売者の対応決済方法を検証するように調整
- 返金取引開始時に、必ず決済取引から決済サービスタイプを取得するように調整
- コンテンツ名称を多言語対応

### Fixed

- 取引番号発行時のエラーハンドリング調整

## v15.45.0 - 2021-02-18

### Changed

- クレジットカード決済サービスの全パラメータにサイト情報をオプションとして追加

## v15.44.0 - 2021-02-05

### Added

- 予約通知先のグローバル設定を追加

### Changed

- COAオファーインポート処理にbulkWriteを使用するように調整

## v15.43.0 - 2021-01-27

### Added

- CreateReservationReportタスクを追加
- SendEmailMessageタスクを追加
- プロジェクトサービスを追加

### Changed

- アクション検索条件拡張
- イベントに入場ゲート集計属性を追加
- ID指定されたイベントについてはEventScheduledでなくても予約集計するように調整
- 予約にreservedTicket.dateUsed属性を実験的に追加
- 予約検索に$projectionを指定できるように調整
- update packages
- update typescript

## v15.42.5 - 2020-12-13

### Changed

- イベントのオファー集計にオファーカテゴリー情報を追加

## v15.42.4 - 2020-12-10

### Changed

- ポイント特典入金前に既に入金済かどうかを確認するように調整

## v15.42.3 - 2020-12-09

### Changed

- ムビチケ返金失敗時のエラーメッセージに情報追加

## v15.42.2 - 2020-12-09

### Changed

- ムビチケ返金失敗時のエラーメッセージに情報追加

## v15.42.1 - 2020-12-09

### Changed

- ポイント特典インターフェースを拡張

## v15.42.0 - 2020-12-08

### Changed

- Pecorino取引に対して識別子を指定できるように調整
- ポイント特典インターフェースを拡張

## v15.41.0 - 2020-12-04

### Changed

- 外部決済サービス認証情報を販売者から決済サービスへ移行
- 販売者検索の条件からpaymentAcceptedを除外
- プロダクト検索時にprojectionを指定できるように調整

## v15.40.1 - 2020-11-24

### Changed

- 座席に名称を追加

## v15.40.0 - 2020-11-24

### Changed

- ルームに検索条件を定義
- ルームセクションに検索条件を定義
- 座席に名称を追加
- オファー検索条件拡張

## v15.39.0 - 2020-11-24

### Changed

- カテゴリーコード検索条件拡張
- 勘定科目コレクションにインデックス追加

## v15.38.0 - 2020-11-21

### Changed

- オファー検索条件拡張

## v15.37.0 - 2020-11-20

### Changed

- カテゴリーコード検索条件拡張
- イベントシリーズ検索条件拡張

## v15.36.0 - 2020-11-20

### Changed

- 作品検索条件拡張
- 座席検索条件拡張
- オファーカタログ検索条件拡張

## v15.35.2 - 2020-11-19

### Changed

- update @chevre/factory

## v15.35.1 - 2020-11-17

### Changed

- サービス登録時に、オファーのポイント特典のtypeOf設定を反映するように調整

## v15.35.0 - 2020-11-17

### Changed

- AccountプロダクトタイプをPaymentCardに統合

## v15.34.0 - 2020-11-15

### Changed

- AccountプロダクトとPaymentCardプロダクトの挙動を統一

## v15.33.0 - 2020-11-13

### Changed

- 決済取引をPaymentCardサービスに対応

## v15.32.0 - 2020-11-11

### Added

- USE_MOVIETICKET_AUTHORIZE設定を追加

## v15.31.0 - 2020-11-04

### Changed

- プロダクトインターフェースを決済サービスにも拡張
- 外部決済サービス認証情報をproductsコレクションから取得するように調整
- ペイメントカードの通貨をPecorinoに反映するように調整
- update @pecorino/api-nodejs-client

## v15.30.1 - 2020-10-29

### Changed

- 勘定科目コレクションのインデックス調整

## v15.30.0 - 2020-10-29

### Changed

- プロダクト検索条件拡張

## v15.29.1 - 2020-10-18

### Changed

- categoryCodesに適切なインデックスを追加

## v15.29.0 - 2020-10-14

### Changed

- 販売者コレクションのインデックスを最適化
- イベントオファーごとのキャパシティ算出時に適用サブ予約条件を考慮するように調整

## v15.28.1 - 2020-10-12

### Changed

- 販売者検索条件から不要なものを削除

## v15.28.0 - 2020-10-10

### Changed

- 販売者検索条件拡張

## v15.27.0 - 2020-10-09

### Changed

- 施設検索条件拡張

## v15.26.1 - 2020-10-02

### Changed

- COAのflgMvtkUseをMGチケットにも適用するように調整

## v15.26.0 - 2020-09-30

### Added

- 決済取引をPaymentServiceType:Accountに対応
- 返金取引をPaymentServiceType:Accountに対応

### Changed


src/util/film.ts
// tslint:disable:max-line-length no-irregular-whitespace
/**
 * 部門リストを取得する
 */
export function getSections() {
    return [
        { code: '01', name: { ja: 'オープニング作品', en: 'Opening' } },
        { code: '02', name: { ja: 'クロージング作品', en: 'Closing' } },
        { code: '03', name: { ja: '授賞式＋東京グランプリ受賞作品上映', en: 'Tokyo Grand-Prix Winner Screening + Award Ceremony' } },
        { code: '04', name: { ja: '特別招待作品', en: 'Special Screening' } },
        { code: '05', name: { ja: '特別上映', en: 'Special Program' } },
        { code: '06', name: { ja: 'ワールド・フォーカス', en: 'World Focus' } },
        { code: '07', name: { ja: 'コンペティション', en: 'Competition' } },
        { code: '08', name: { ja: '東京グランプリ受賞作品', en: 'Tokyo Grand-Prix Winner Screening' } },
        { code: '09', name: { ja: '観客賞受賞作品', en: 'Audience Award Winner Screening' } },
        { code: '10', name: { ja: 'アジアの未来', en: 'Asian Future' } },
        { code: '11', name: { ja: '日本映画スプラッシュ', en: 'Japanese Cinema Splash' } },
        { code: '12', name: { ja: '国際交流基金アジアセンター presents CROSSCUT ASIA　＃03 カラフル！ インドネシア', en: 'The Japan Foundation Asia Center presents CROSSCUT ASIA  #03 COLORFUL INDONESIA' } },
        { code: '13', name: { ja: 'アニメーション特集　「映画監督　細田守の世界」', en: 'Special Focus on Japanese Animation: The World of Mamoru Hosoda' } },
        { code: '14', name: { ja: 'Japan Now', en: 'Japan Now' } },
        { code: '15', name: { ja: 'Japan Now「監督特集　岩井俊二」', en: 'Japan Now Director in Focus: Shunji Iwai' } },
        { code: '16', name: { ja: '日本映画クラシックス', en: 'Japanese Classics' } },
        { code: '17', name: { ja: 'アジア・オムニバス映画製作シリーズ アジア三面鏡2016：リフレクションズ', en: 'Asian Three-Fold Mirror 2016: Reflections' } },
        { code: '18', name: { ja: 'みなと上映会', en: 'Minato Screening' } },
        { code: '19', name: { ja: '日本映画監督協会新人賞、上映とシンポジウム', en: 'Directors Guild of Japan New Directors Award Film Screening and Symposium' } },
        { code: '20', name: { ja: 'PFF（ぴあフィルムフェスティバル）「PFFアワード2016」グランプリ受賞作品上映', en: '2016 PFF Award-Winner Film Screening' } },
        { code: '21', name: { ja: '「SKIPシティ国際Dシネマ映画祭2016」作品上映', en: 'Screening of SKIP CITY INTERNATIONAL D-Cinema FESTIVAL 2016' } },
        { code: '22', name: { ja: 'ユース（TIFFティーンズ）', en: 'Youth ( TIFF-Teens)' } },
        { code: '23', name: { ja: 'ウルトラマンシリーズ放送開始50年　特別企画：脚本家 金城哲夫', en: '50 years since broadcasting of the ULTRAMAN series Special Program: Scriptwriter Tetsuo Kinjo' } },
        { code: '24', name: { ja: 'WOWO映画工房×ジャパンプレミア feat. スター・ウォーズ in 東京国際映画祭', en: 'All night screening presented by WOWOW\'s "CINEMA STUDIO" featuring Star Wars documentary' } },
        { code: '25', name: { ja: '日本映画スプラッシュ・場外乱闘編（アウト・オブ・コンペティション）', en: 'Japanese Cinema Splash Out of Competition' } },
        { code: '26', name: { ja: '第６回日本学生映画祭', en: 'The 6th Japan Student Film Festival' } }
    ];
}
